### DISCUSSION TITLE
CTRL+F save the last search

### DISCUSSION THREAD

**Discussion by batata004** [microsoft/monaco-editor#3743] [Q&A] [answered]
Sometimes I need to open the Monaco Editor for several files and search a specific word like `xxx`. So everytime I load the monaco editor I need to press `CTRL+ F` -> wait the find box appears and type `xxx` so it can search.

I can easily save what the user searches at some cookie  Then my question is this: is there an event that is triggered as soon as the search box appears so I can fill its input with the value I saved on my cookie?

**Comment by alexdima** [comment, marked as answer]
@batata004 We don't have that in the API of the editor, but this being JS you could try something like the following (paste at https://microsoft.github.io/monaco-editor/playground.html and Run):


```js
const editor = monaco.editor.create(document.getElementById("container"), {
	value: "function hello() {\n\talert('Hello world!');\n}",
	language: "javascript"
});
editor.getContribution('editor.contrib.findController')._state.onFindReplaceStateChange((e) => {
    // console.log(e);
    if (e.isRevealed) {
        console.log(`find replace is now visible`);
    }
})
```

**Comment by batata004** [comment]
@alexdima thank you for helping! And if you could take a look at https://github.com/microsoft/monaco-editor/issues/2410#issuecomment-832107194 because you can see the suggested solution in that issue worked but to me it looks like a hack. Do you know if is possible to trigger the find/replace action?
